src/lib/utils/track.ts

```diff
@@ -0,0 +1,107 @@
+import * as React from "react"
+import { track as _track, Track as _Track } from "react-tracking"
+
+// tslint:disable-next-line:no-namespace
+export namespace Schema {
+  /**
+   * The global tracking-info keys in Artsy’s schema.
+   */
+  export interface Global {
+    /**
+     * The name of an event.
+     */
+    action: string
+
+    /**
+     * The root container component should specify this as the screen context.
+     */
+    page: string
+  }
+
+  export interface Entity extends Global {
+    /**
+     * The ID of the entity in its database. E.g. the Mongo ID for entities that reside in Gravity.
+     */
+    entity_id: string
+
+    /**
+     * The public slug for this entity.
+     */
+    entity_slug: string
+  }
+}
+
+/**
+ * Use this interface to augment the `track` function with props, state, or custom tracking-info schema.
+ *
+ * @example
+ *
+ *      ```ts
+ *      import { Schema, Track, track as _track } from "src/utils/track"
+ *
+ *
+ *      interface Props {
+ *        artist: {
+ *          id: string
+ *          slug: string
+ *        }
+ *      }
+ *
+ *      interface State {
+ *        following: boolean
+ *      }
+ *
+ *      const track: Track<Props, State, Schema.Entity> = _track
+ *
+ *      @track()
+ *      class Artist extends React.Component<Props, State> {
+ *        render() {
+ *          return (
+ *            <div onClick={this.handleFollow.bind(this)}>
+ *              ...
+ *            </div>
+ *          )
+ *        }
+ *
+ *        @track((props, state) => ({
+ *          action: `${state.following ? "Unfollow" : "Follow"} Artist`,
+ *          entity_id: props.artist.id,
+ *          entity_slug: props.artist.slug
+ *        }))
+ *        handleFollow() {
+ *          // ...
+ *        }
+ *      }
+ *
+ *      ```
+ */
+export interface Track<P = any, S = any, T extends Schema.Global = Schema.Global> extends _Track<T, P, S> {} // tslint:disable-line:no-empty-interface
```

Comment:
I'm confused here (not sure if this is what Luc was referring to above as well)...

So `Track` takes `P, S, T`, but `_Track` takes `T, P, S`?

Reply:
lol, yup, you’re correct and that’s what @l2succes was asking about.

Reply:
👍 I think it makes sense. Just had to take a sec to make sure haha 😄